### PR TITLE
Robot connect

### PR DESCRIPTION
**Pull request by Paladinking**
Working robot communication, pinging + synchronization using tx_semaphore.

### CODE REVIEW

File: .mxproject [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
...x\common\src\tx_thread_sleep.c;Middlewares\ST\threadx\common\src\tx_thread_stack_analyze.c;Middlewares\ST\threadx\common\src\tx_thread_suspend.c;Middlewares\ST\threadx\common\src\tx_thread_system_preempt_check.c;Middlewares\ST\threadx\common\src\tx_thread_system_suspend.c;Middlewares\ST\threadx\common\src\tx_thread_terminate.c;Middlewares\ST\threadx\common\src\tx_thread_time_slice.c;Middlewares\ST\threadx\common\src\tx_thread_time_slice_change.c;Middlewares\ST\threadx\common\src\tx_thread_timeout.c;Middlewares\ST\threadx\common\src\tx_thread_wait_abort.c;Middlewares\ST\threadx\common\src\tx_time_get.c;Middlewares\ST\threadx\common\src\tx_time_set.c;Middlewares\ST\threadx\common\src\txe_block_allocate.c;Middlewares\ST\threadx\common\src\txe_block_pool_create.c;Middlewares\ST\threadx\common\src\txe_block_pool_delete.c;Middlewares\ST\threadx\common\src\txe_block_pool_info_get.c;Middlewares\ST\threadx\common\src\txe_block_pool_prioritize.c;Middlewares\ST\threadx\common\src\txe_block_release.c;Middlewares\ST\threadx\common\src\txe_byte_allocate.c;Middlewares\ST\threadx\common\src\txe_byte_pool_create.c;Middlewares\ST\threadx\common\src\txe_byte_pool_delete.c;Middlewares\ST\threadx\commo
```
Comment (r0x57n) [inline .mxproject:13]: ouggggh

File: AZURE_RTOS/App/app_azure_rtos_config.h [hunk 2]
```diff
@@ -43,7 +43,7 @@ extern "C" {
 
 #define USE_STATIC_ALLOCATION                1
 
-#define TX_APP_MEM_POOL_SIZE                     1024
+#define TX_APP_MEM_POOL_SIZE                     4096
```
Comment (r0x57n) [inline AZURE_RTOS/App/app_azure_rtos_config.h:46]: visst behövdes detta i slutändan, inte bara nånting vi glömde kvar?

**Review by r0x57n — CHANGES_REQUESTED** [review]
Kolla kommentarer, men utöver dom så kan jag inte bygga:
/home/re/dev/seegoals/basestation/Core/Src/com.c:5:10: fatal error: nrf_helper_defines.h: No such file or directory
    5 | #include <nrf_helper_defines.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/basestation.elf.dir/build.make:104: CMakeFiles/basestation.elf.dir/Core/Src/com.c.obj] Error 1
make[2]: *** Waiting for unfinished jobs....
/home/re/dev/seegoals/basestation/Core/Src/handle_packet.c:3:10: fatal error: robot_action/robot_action.pb-c.h: No such file or directory
    3 | #include <robot_action/robot_action.pb-c.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

File: AZURE_RTOS/App/app_azure_rtos_config.h (hunk 2 above)
Comment (Paladinking) [reply AZURE_RTOS/App/app_azure_rtos_config.h:46]: Ja, det behövs, 1024 var för lite.

**Comment by Paladinking** [comment]
> Kolla kommentarer, men utöver dom så kan jag inte bygga: /home/re/dev/seegoals/basestation/Core/Src/com.c:5:10: fatal error: nrf_helper_defines.h: No such file or directory 5 | #include <nrf_helper_defines.h> | ^~~~~~~~~~~~~~~~~~~~~~ compilation terminated. make[2]: *** [CMakeFiles/basestation.elf.dir/build.make:104: CMakeFiles/basestation.elf.dir/Core/Src/com.c.obj] Error 1 make[2]: *** Waiting for unfinished jobs.... /home/re/dev/seegoals/basestation/Core/Src/handle_packet.c:3:10: fatal error: robot_action/robot_action.pb-c.h: No such file or directory 3 | #include <robot_action/robot_action.pb-c.h> | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ compilation terminated.

Jag lyckas inte upprepa detta, gjorde en build på en dator som inte hade något installerat och det fungerade.
Var dina submoduler updaterade rätt @r0x57n ?

**Review by r0x57n — APPROVED** [review]

**Pull request merged**